src/client/Client.cc

```diff
@@ -6934,8 +6979,27 @@ int Client::_do_setattr(Inode *in, struct ceph_statx *stx, int mask,
     if (mask & (CEPH_SETATTR_MTIME|CEPH_SETATTR_ATIME)) {
       if (mask & CEPH_SETATTR_MTIME)
         in->mtime = utime_t(stx->stx_mtime);
-      if (mask & CEPH_SETATTR_ATIME)
-        in->atime = utime_t(stx->stx_atime);
+      if (mask & CEPH_SETATTR_ATIME) {
+        uint32_t min_atime_sec = in->mtime.sec()+ in->worm.min_retention_period;
```

Comment:
Since the acquisition time comes from the local, there is a modification of the local time to cause the folder to enter the protection period in advance. It may be provided by providing an external standard clock source interface or in the document: "Synchronize the time before setting the WORM function"

What do you think? @ukernel @batrick @jtlayton

Reply:
I think it'd be better to not overload the atime field. While it's not terribly common, some applications _do_ use it. Why not store this in a separate field in the inode and present it via xattr?

Reply:
OK. Already drop to use atime and add a new field in worm info struct insteed.